### ISSUE TITLE
upload-to-f4040.sh ftp issues

### ISSUE THREAD

**Issue by mwarning** [chunkeey/FritzBox-4040-UBOOT#5]
Some FTP versions have different settings:
-  passive mode is the default. The `passive` command in the `upload-to-f4040.sh` script then toggles and disables passive mode.
- the default timeout is too low. It takes 2 minutes until the data transfer starts after the put command.

This has been seen on  tnftp-20151004_8 (linked as ftp command) on Void Linux.
The solution for me was to `ftp -p -q 150 ...`.

**Comment by blocktrron** [comment]
Hmm, i guess most people use https://github.com/freifunk-darmstadt/fritz-tools today (Might be biased as I've written this tool).

Also, this is what i would recommend people to use (`upload-to-f4040.sh` predates fritzflash.py). Maybe the README should be adjusted to link to this tool?

**Comment by mwarning** [comment]
OpenWrt master uses this as source (openwrt/package/boot/uboot-fritz4040), so that led me here. :-)

**Comment by blocktrron** [comment]
You are right, OpenWrt packages the upload script alongside the bootloader image. From my perspective, this is the wrong approach, as OpenWrt itself has a flashing script directory within the project.

So removing the flashing script and linking the aforementioned tool (or alternatively add a more robust shell implementation to OpenWrt) should be the way to go. What do you think?

**Comment by mwarning** [comment]
It depends, 
```
$ ls bar/openwrt/bin/targets/ipq40xx/generic/u-boot-fritz4040/
uboot-fritz4040.bin  upload-to-f4040.sh
```
I do not quite understand what this bin file is for. It is flashed with the same commands (same partition) as regular sysupgrade images. I tried to flash it, but the device does not seem to boot up.

Probably whole folder can be taken away.

**Comment by blocktrron** [comment]
This file was part of the two-step installation process used prior to the single EVA image which was added later.

**Comment by mwarning** [comment]
Then the whole folder can be removed...

**Comment by blocktrron** [comment]
This requires a per-botloader target decision, as other models built from the uboot-fritz4040 package require the raw bootloader for successful flashing (Basically all other models than the 4040).

**Comment by mwarning** [comment]
I that case, the image should be left. But I think the script can be removed. Some documentation what that images does might be useful to have (in the wiki)
